python/graphstorm/model/edge_decoder.py

```diff
@@ -487,17 +487,64 @@ def _compute_logits(self, g, h):
         return out
 
 ##################### Link Prediction Decoders #######################
-class LinkPredictDotDecoder(GSLayerNoParam):
-    """ Link prediction decoder with the score function of dot product
+class LinkPredictNoParamDecoder(GSLayerNoParam):
```

Comment:
can you explain why we need to distinguish them?

Reply:
LinkPredictNoParamDecoder Is for decoders that do not have trainable parameters such as LinkPredictDotDecoder.
LinkPredictLearnableDecoder is for decoders that have trainable parameters, i.e., nn.Module.

That's why we need to define two different abstract classes.